axelrod/strategies/adaptor.py

```diff
@@ -0,0 +1,106 @@
+from typing import Dict, Tuple
+
+from axelrod.action import Action
+from axelrod.player import Player
+from axelrod.random_ import random_choice
+
+from numpy import heaviside
+
+C, D = Action.C, Action.D
+
+
+class AbstractAdaptor(Player):
+    """
+    An adaptive strategy that updates an internal state based on the last
+    round of play. Using this state the player Cooperates with a probability
+    derived from the state.
+
+    Parameters
+    ----------
```

Comment:
The `----...` need to be omitted for strategy docstrings (our setup can't handle them).